cmd/podman/create.go

```diff
@@ -77,6 +77,16 @@ func createInit(c *cliconfig.PodmanCommand) error {
 		logrus.Warn("setting security options with --privileged has no effect")
 	}
 
+	var setNet string
+	if c.IsSet("network") {
+		setNet = c.String("network")
+	} else if c.IsSet("net") {
+		setNet = c.String("net")
+	}
+	if (c.IsSet("dns") || c.IsSet("dns-opt") || c.IsSet("dns-search")) && setNet != "" && setNet != "host" {
```

Comment:
I think this should be blocked if setNet == NONE, and CONTAINER:?

Reply:
So it should return error
`if (c.IsSet("dns") || c.IsSet("dns-opt") || c.IsSet("dns-search")) && (setNet == "none" || setNet == "host" || strings.HasPrefix("container:")`
because it makes no sense to set /etc/resolv.conf if `--network=host` or `--network=none`?

Reply:
DNS configuration options are only disallowed when `--net=container:$id` is set. Other values (host, bridge, etc) seem fine. `--net=none` is a possible exception - I don't think it matters, so we can either allow or deny.

Reply:
If `none` doesn't matter, I choose to allow `none` and docker doesn't block `none` either.

There's another kind of --network `'ns:<path>': path to a network namespace to join`, will this be influenced with --dns*?

To only disallow if `--net container:id` or `--network container:id`
```
if (c.IsSet("dns") || c.IsSet("dns-opt") || c.IsSet("dns-search")) && setNet != "" && strings.HasPrefix(setNet, "container:")
```

Reply:
Oh, `--net=ns:` should also disallow DNS options - good catch

Reply:
Shouldn't we allow dns with `ns:`?  I'd say it is a valid combination

Reply:
`ns` attempts to grab /etc/resolv.conf and /etc/hosts from a special path... somewhere in /run, I think? On the assumption that 'ip netns create' made the namespace (it also makes resolv.conf and hosts).